glusterd2/commands/volumes/volume-edit.go

```diff
@@ -60,6 +62,14 @@ func volumeEditHandler(w http.ResponseWriter, r *http.Request) {
 		if req.DeleteMetadata {
 			delete(volinfo.Metadata, key)
 		} else {
+			offset := len(key) + len(value)
```

Comment:
The variable name `offset` is a little misleading. What do you think of below approach ?


Check `if req.MetadataSize()` is more than metadata size limit
If yes, return error.

If not, loop over and update `volinfo.Metadata`.
Now check if `volinfo.MetadataSize()` is more than metadata size limit.